utilities/transactions/write_unprepared_txn.h

```diff
@@ -25,15 +25,16 @@ class WriteUnpreparedTxnReadCallback : public ReadCallback {
                                  WriteUnpreparedTxn* txn)
       // Disable snapshot check on parent class since it would violate
       // read-your-own-write guarantee.
-      : ReadCallback(kMaxSequenceNumber, min_uncommitted),
+      : ReadCallback(kMaxSequenceNumber, min_uncommitted,
+                     CalcMaxUnpreparedSequenceNumber(txn)),
```

Comment:
So the reason why max unprep seq no isn't calculated just once is because a transaction can accumulate more unprep_seqs.

I think that that is a real possibility right? You could be writing while iterating through a DB (eg. iterate through all keys and issuing Delete for them). One way is for transactions to have back pointers into readcallbacks, so that we can just update them.

Reply:
If that is the case, how come it is not protected against concurrent access via a mutex?

Reply:
Looking around, I don't think we support concurrent access to the same transaction object (eg. WriteBatchWithIndex is modified w/o locking).

The issue is just interleaving reads/writes from the same transaction.